wire-runtime/src/commonMain/kotlin/com/squareup/wire/Message.kt

```diff
@@ -0,0 +1,41 @@
+package com.squareup.wire
+
+import okio.BufferedSink
+import okio.ByteString
+
+/** A protocol buffer message. */
+expect abstract class Message<M : Message<M, B>, B : Message.Builder<M, B>> {
+  /** If non-zero, the hash code of this message. Accessed by generated code. */
+  protected var hashCode: Int
+
+  /**
+   * Returns a byte string containing the proto encoding of this message's unknown fields. Returns
+   * an empty byte string if this message has no unknown fields.
+   */
+  fun unknownFields(): ByteString
+
+  /**
+   * Returns a new builder initialized with the data in this message.
+   */
+  // TODO(egorand): Remove from common and not generate in Kotlin.
```

Comment:
Hmm, doesn't seem feasible, `RuntimeMessageAdapter` still needs to reference it in a generic way.